src/main.rs

```diff
@@ -189,6 +218,12 @@ Report issues to <https://github.com/lunaryorn/mdcat>.",
                 .help("Dump Markdown parser events and exit")
                 .hidden(true),
         )
+        .arg(
+            Arg::with_name("fail_fast")
+                .long("fail-fast")
+                .help("Exit if any error occurs processing an input file")
+                .hidden(true),
+        )
```

Comment:
I'd prefer if we did not add this flag.  Why can't we just follow `cat`'s behaviour (which doesn't have this flag either)?  `cat`'s been like this for >20 years, so it can't be too wrong :)

Reply:
My use case for mdcat would be easier with the flag and my reason for making this change in the first place.   If it just a preference, can we just leave this change in?

Reply:
What is your use case?  

Why do you need it in `mdcat` itself? After all you could just invoke `mdcat` repeatedly, as in `for file in files; do mdcat "$file" || exit 1; done`.

As said, `cat`'s got no such flag, and I'd like a good reason for making `mdcat` behave differently, and also as to why I need to maintain both behaviours.

Reply:
Thinking of it the flag doesn't do any harm either, so we can keep it, but then let's shorten it to `--fail`, make it public, add a description and add it to the changelog.

Reply:
It is the concatenation feature and knowing which markdown segment (file) is missing or does not parse cleanly within a pipeline.    (mdcat ${var[@]} |  prog1)

If something is missing, we do not want the files after the error to show up to prog1

Reply:
As said, I'm fine with the flag if we make it "official", but concerning your example, wouldn't

```bash
for file in "${var[@]}"; do 
  mdcat "$file" || exit 1
done | prog1
```
have the same effect?

Reply:
I see you understand the bash shell loop termination issue, but we still want prog1 to run with the input it has and not just die.   Breaking out of the loop instead causes timing issues if the data in the pipeline is processed or not.

There is also an overhead performance penalty if you have to execute mdcat N times instead of just once.

Reply:
Thanks :)